crates/core/app/tests/spend.rs

```diff
@@ -96,7 +96,7 @@ async fn spend_happy_path() -> anyhow::Result<()> {
 #[tokio::test]
 #[cfg_attr(
     debug_assertions,
-    should_panic("assertion failed: cs.is_satisfied().unwrap()")
+    should_panic = "assertion failed: cs.is_satisfied().unwrap()"
```

Comment:
Not a huge deal, but reading the last state of the proof test, I am not sure I understand why we expect this to panic? the attribute makes me think it might be a debug assert nested deep in arkworks? If so we should probably add a comment

Reply:
This is cursed, if you try to verify an invalid proof _in debug mode_, arkworks will panic with this error. So we expect this to panic in debug mode, in release mode it should error with an invalid spend proof. The way the test is written currently with contains() is a bug